### ISSUE TITLE
Add `: Display` as a supertrait for `AstNode`

### ISSUE THREAD

**Issue by matklad** [rust-analyzer/rust-analyzer#3486]
Printing `AstNode`s is awkward, as one needs to call `.syntax()` on them. We should add `Display` as a super-trait to AstNode [here](https://github.com/rust-analyzer/rust-analyzer/blob/d75577fcee79aac06bdddb01fde431f26432c36c/crates/ra_syntax/src/ast.rs#L33) and make sure that all nodes actually implement this, by tweaking codegen [here](https://github.com/rust-analyzer/rust-analyzer/blob/d75577fcee79aac06bdddb01fde431f26432c36c/xtask/src/codegen/gen_syntax.rs#L27)

It should look like

```rust
impl fmt::Display for Foo {
    fn fmt(..) { fmt::Display::fmt(self.syntax()) }
}
```

it probably makes sense to stick all Dispaly impls at the end of the file, to make the meaninful impls less cluttered.

**Comment by calvinbrown085** [comment]
@matklad I can pick this one up!

**Comment by calvinbrown085** [comment]
@matklad So, before I write to much code we'd want to do something like,

```
impl fmt::Display for AstNode {
    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {

    }
}
```

in ast.rs? https://github.com/rust-analyzer/rust-analyzer/blob/d75577fcee79aac06bdddb01fde431f26432c36c/crates/ra_syntax/src/ast.rs#L33

**Comment by matklad** [comment]
@calvinbrown085 you can't implement traits directy on traits (and both `Dispaly` and `AstNode` are traits). It's also not possible to write a "blanket impl" in this case, as stdlib already has a blanker impl for `&T where T: Display`. So, we need to generate a seaprate impl block for each type that implements AstNode in the [codegen](https://github.com/rust-analyzer/rust-analyzer/blob/d75577fcee79aac06bdddb01fde431f26432c36c/xtask/src/codegen/gen_syntax.rs#L27)

**Comment by simonvandel** [comment]
@matklad can this be closed now that #3562 is merged?
